### PR TITLE
Global : Enable room sharing feature and room access by link feature

### PR DESCRIPTION
**Pull request by Yoshiin**
This PR is strongly linked to #329 

Resolve dinsic-pim/tchap-web#120

### CODE REVIEW

**Comment by Yoshiin** [comment]
Some things to change : 
- [ ] Use the "Element" sharing field
- [ ] Show a disabled version of the switches ("External" and "Link") for the non-admin users
- [ ] Add a dialog in order to hook errors returned by the server

File: src/components/views/room_settings/RoomProfileSettings.js [hunk 1]
```diff
@@ -155,37 +173,191 @@ export default class RoomProfileSettings extends React.Component {
...
+                });
+            }).catch(err => {
+                console.error(err);
+            });
+        } else {
+            this._setJoinRules(room, "public");
+        }
+    };
+
+    _onLinkSharingSwitchChange = (e) => {
```
Comment (giomfo) [inline src/components/views/room_settings/RoomProfileSettings.js:276]: "e" doesn't mean really something for me

File: src/components/views/rooms/RoomHeader.js [hunk 2]
```diff
@@ -266,6 +274,19 @@ module.exports = React.createClass({
                 </AccessibleButton>;
         }
 
+        let shareRoomButton;
+        if (this.props.inRoom && (Tchap.isRoomForum(this.props.room) ||
+            (MatrixClientPeg.get().isRoomEncrypted(this.props.room.roomId)
+                && Tchap.getJoinRules(this.props.room) === "public"
+                && Tchap.getAccessRules(this.props.room.roomId) === "restricted"))) {
```
Comment (giomfo) [inline src/components/views/rooms/RoomHeader.js:281]: Don't you think this condition `Tchap.getAccessRules(this.props.room.roomId) === "restricted"` should be removed?

File: src/i18n/strings/fr.json [hunk 3]
```diff
@@ -1958,5 +1959,13 @@
     "Stay connected from at least one other device": "Restez connecté depuis au moins un autre appareil",
     "Export your encryption keys": "Exporter vos clés de chiffrement",
     "Find out more": "En savoir plus",
-    "Export keys and log-out": "Exporter les clés et se déconnecter"
+    "Export keys and log-out": "Exporter les clés et se déconnecter",
+    "Activate link access to this room": "Activer l'accès au salon par lien",
+    "Users can join this room with the following link:": "Les utilisateurs peuvent rejoindre ce salon à partir du lien suivant :",
+    "Failed to open link access for this room": "Échec de l'ouverture de l'accès par lien à ce sallon",
```
Comment (giomfo) [inline src/i18n/strings/fr.json:1965]: typo "sallon"

File: src/i18n/strings/fr.json [hunk 4]
```diff
@@ -1958,5 +1959,13 @@
...
+    "This change is not currently supported because externs are allowed to join this room.": "Ce changement n'est pas supporté actuellement car les externes sont autorisés à rejoindre ce salon.",
+    "Failed to open this room to externs": "Échec de l'ouverture aux externes de ce sallon",
```
Comment (giomfo) [inline src/i18n/strings/fr.json:1967]: typo "sallon"

File: src/components/views/room_settings/RoomProfileSettings.js [hunk 5]
```diff
@@ -155,37 +173,191 @@ export default class RoomProfileSettings extends React.Component {
...
+        const self = this;
+        client.sendStateEvent(room.roomId, "m.room.join_rules", { join_rule: joinRules }, "").then(() => {
+            self.setState({
+                link_sharing: joinRules === "public",
+                joinRules,
+            });
+        }).catch((err) => {
+            console.error(err);
+            this.setState({
+                link_sharing: false,
```
[the remark this answers is not in the document]
Comment (Yoshiin) [reply src/components/views/room_settings/RoomProfileSettings.js:238]: It simply can be removed

File: src/components/views/room_settings/RoomProfileSettings.js (hunk 1 above)
Comment (Yoshiin) [reply src/components/views/room_settings/RoomProfileSettings.js:276]: "e" like "event". Will be updated.

File: src/components/views/rooms/RoomHeader.js (hunk 2 above)
Comment (Yoshiin) [reply src/components/views/rooms/RoomHeader.js:281]: Yes !

File: src/components/views/rooms/RoomHeader.js [hunk 6]
```diff
@@ -266,6 +277,18 @@ module.exports = React.createClass({
                 </AccessibleButton>;
         }
 
+        let shareRoomButton;
+        if (this.props.inRoom && (Tchap.isRoomForum(this.props.room) ||
+            (MatrixClientPeg.get().isRoomEncrypted(this.props.room.roomId)
+                && Tchap.getJoinRules(this.props.room) === "public"))) {
```
Comment (giomfo) [inline src/components/views/rooms/RoomHeader.js:283]: Maybe you should just check "(this.props.inRoom && Tchap.getJoinRules(this.props.room) === "public)" here

**Review by giomfo — APPROVED** [review]
SGTM, just one suggestion

File: src/components/views/rooms/RoomHeader.js (hunk 6 above)
Comment (Yoshiin) [reply src/components/views/rooms/RoomHeader.js]: Yes, true.

**Pull request merged**